### ISSUE TITLE
Regression: Gradle plugin requires GRAALVM_HOME to be set

### ISSUE THREAD

**Issue by sdeleuze** [graalvm/native-build-tools#109]
I think this is a regression we previously saw and fix with @melix but it seems to be back with 0.9.2. When using https://sdkman.io to install and use GraalVM, `GRAALVM_HOME` env variable is not set and when using the Gradle plugin and I run `./gradlew clean nativeBuild` I get the following error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeBuild'.
> Expected to find native-image executable but it wasn't found. Make sure to declare the GRAALVM_HOME environment variable or install GraalVM with native-image in a standard location recognized by Gradle Java toolchain support
```

The same use case works correctly with 0.9.1. If I set `GRAALVM_HOME` with 0.9.2 it works.

**Comment by lazar-mitrovic** [comment]
Did you install `native-image` using `gu install native-image`?

**Comment by sdeleuze** [comment]
Yes I did. After more tests, I found this regression can be reproduced with Gradle 6.8.3, not with Gradle 7.x.

**Comment by lazar-mitrovic** [comment]
I cannot reproduce this locally with Gradle 6.8.3:
```bash
$ cd ~/Desktop/native-image-build-tools/samples/java-application
$ cat gradle/wrapper/gradle-wrapper.properties 
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-bin.zip
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
$ which native-image
~/.sdkman/candidates/java/current/bin/native-image
$ ./gradlew clean nativeBuild

> Task :generateResourcesConfigFile
[native-image-plugin] Resources configuration written into ~/Desktop/native-image-build-tools/samples/java-application/build/native/generated/generateResourcesConfigFile/resource-config.json

> Task :nativeBuild
[java-application:19565]    classlist:   1,015.79 ms,  0.96 GB
[java-application:19565]        (cap):     409.29 ms,  0.96 GB
[java-application:19565]        setup:   1,697.21 ms,  0.96 GB
[java-application:19565]     (clinit):     174.56 ms,  1.75 GB
[java-application:19565]   (typeflow):   3,704.35 ms,  1.75 GB
[java-application:19565]    (objects):   2,918.59 ms,  1.75 GB
[java-application:19565]   (features):     312.97 ms,  1.75 GB
[java-application:19565]     analysis:   7,323.64 ms,  1.75 GB
[java-application:19565]     universe:     562.10 ms,  1.76 GB
[java-application:19565]      (parse):     582.19 ms,  1.76 GB
[java-application:19565]     (inline):   1,080.49 ms,  2.66 GB
[java-application:19565]    (compile):   7,035.75 ms,  4.55 GB
[java-application:19565]      compile:   9,422.63 ms,  4.55 GB
[java-application:19565]        image:   1,663.53 ms,  4.55 GB
[java-application:19565]        write:     208.01 ms,  4.55 GB
[java-application:19565]      [total]:  22,099.55 ms,  4.55 GB
# Printing build artifacts to: ~/Desktop/native-image-build-tools/samples/java-application/build/native/nativeBuild/java-application.build_artifacts.txt
[native-image-plugin] Native Image written to: ~/Desktop/native-image-build-tools/samples/java-application/build/native/nativeBuild

BUILD SUCCESSFUL in 23s
5 actionable tasks: 5 executed
```
I attached [the diff used](https://github.com/graalvm/native-build-tools/files/6925753/patch.txt).

Can you run `gradle -q javaToolchains`? On my machine it returns:
```bash
$ ./gradlew -q javaToolchains

 + Options
     | Auto-detection:     Enabled
     | Auto-download:      Enabled

 + AdoptOpenJDK 1.8.0_292
     | Location:           ~/.gradle/jdks/jdk8u292-b10
     | Language Version:   8
     | Vendor:             AdoptOpenJDK
     | Is JDK:             true
     | Detected by:        Auto-provisioned by Gradle

 + AdoptOpenJDK 15.0.1
     | Location:           ~/.gradle/jdks/jdk-15.0.1+9
     | Language Version:   15
     | Vendor:             AdoptOpenJDK
     | Is JDK:             true
     | Detected by:        Auto-provisioned by Gradle

 + GraalVM Community JDK 11.0.12
     | Location:           ~/.sdkman/candidates/java/21.2.0.r11-grl
     | Language Version:   11
     | Vendor:             GraalVM Community
     | Is JDK:             true
     | Detected by:        Current JVM

 + Ubuntu JRE 11.0.11
     | Location:           /usr/lib/jvm/java-11-openjdk-amd64
     | Language Version:   11
     | Vendor:             Ubuntu
     | Is JDK:             false
     | Detected by:        Common Linux Locations
```

**Comment by darxriggs** [comment]
I **can** reproduce the issue with
- plugin version `0.9.2`
- JDK 8 on the commandline (either `sdk use java 21.2.0.r8-grl` or `sdk use java 8.0.292.hs-adpt`)
- Gradle 6.8.3

I **can't** reproduce the issue with
- plugin version `0.9.2`
- JDK 8 on the commandline (either `sdk use java 21.2.0.r8-grl` or `sdk use java 8.0.292.hs-adpt`)
- Gradle 7.1.1

I have used `gradle wrapper --gradle-version 6.8.3` to generate the Gradle wrapper files and altered the line `id 'org.graalvm.buildtools.native'` to also contain `version '0.9.2'`.


Another observation with Gradle 7.1.1:
With plugin version `0.9.2` and a non-GaalVM (`sdk use java 8.0.292.hs-adpt`) on the command line it even works fine without defining a toolchain that requires GraalVM. How is that?

With plugin version `0.9.1` and a non-GaalVM (`sdk use java 8.0.292.hs-adpt`) on the command line it reports the missing `native-image` which is more expected:
```
> Task :nativeBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeBuild'.
> A problem occurred starting process 'command '~/.sdkman/candidates/java/8.0.292.hs-adpt/bin/native-image''
```

**Comment by melix** [comment]
This is actually a different issue from the one that we fixed, @sdeleuze . Gradle <7 doesn't support setting the vendor for toolchains: https://github.com/graalvm/native-build-tools/blob/44000132425ed03023936830163ceaeff3382050/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java#L214-L216

The consequence is that if you run Gradle 6.8.x with a GraalVM 8 SDK **and** that you didn't change the default language version (11) **and** that you don't have `GRAALVM_HOME` set, then Gradle will try to use (and potentially download) a Java 11 SDK, which will be AdoptOpenJDK 11.

In other words, the default behavior of the plugin is to require a Java 11 release of `native-image`, but what you're running Gradle with is a Java 8 version. Because it needs Java 11, it will use the first Java 11 SDK it finds, which will be the Adopt OpenJDK version.

As a workaround, we could change the default version from 11 to the current Java version _or_ to the `java` extension toolchain version (which might be trickier). Then the current Java version would be deemed compatible, and we would find `native-image`.
